src/Common/Auth/OpenIDConnect/JWT/RsaSha384Signer.php

```diff
@@ -129,17 +128,8 @@ public function verify($expected, $payload, $key): bool
             '  <Modulus>' . $this->b64url2b64($jwk->n) . "</Modulus>\r\n" .
             '  <Exponent>' . $this->b64url2b64($jwk->e) . "</Exponent>\r\n" .
             '</RSAKeyValue>';
-        if (class_exists('Crypt_RSA', false)) {
-            $rsa = new Crypt_RSA();
-            $rsa->setHash(self::CRYPT_ALGORITHM);
-            $rsa->loadKey($public_key_xml, Crypt_RSA::PUBLIC_FORMAT_XML);
-            $rsa->signatureMode = Crypt_RSA::SIGNATURE_PKCS1;
-        } else {
-            $rsa = new \phpseclib\Crypt\RSA();
-            $rsa->setHash(self::CRYPT_ALGORITHM);
-            $rsa->loadKey($public_key_xml, \phpseclib\Crypt\RSA::PUBLIC_FORMAT_XML);
-            $rsa->signatureMode = \phpseclib\Crypt\RSA::SIGNATURE_PKCS1;
-        }
+        $rsa = PublicKeyLoader::load($public_key_xml)->withPadding(RSA::SIGNATURE_PKCS1)->withHash(self::CRYPT_ALGORITHM);
```

Comment:
This RsaSha384Signer.php stuff is testing well in unit testing.
Not sure how to test the SFTP stuff though (per documentation, there appears to be no real changes in use of SFTP in version 3 phpseclib, although would be nice to ensure it tests well :) ).

Reply:
sftp is testing well

Reply:
thanks for testing @stephenwaite !